chainer/functions/connection/linear.py

```diff
@@ -109,7 +132,13 @@ def forward(self, inputs):
                 1 in gy.shape):
             gy = numpy.ascontiguousarray(gy)
 
-        gW = gy.T.dot(x).astype(gy.dtype, copy=False)
+        if self._n_batch_axes > 1:
+            xp = cuda.get_array_module(*inputs)
+            ax = six.moves.range(self._n_batch_axes)
+            gW = xp.tensordot(
+                gy, x, axes=(ax, ax)).astype(gy.dtype, copy=False)
```

Comment:
I think we need to use `W.dtype` here because `W` and `y` can have different dtypes.

Reply:
Thanks! fixed it